Sources/TuistCore/Analytics/CommandEventGraphTarget.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import XcodeGraph
+
+/// A simplified `GraphTarget` to store in `CommandEvent`. As this model gets stored on disk, we want to minimize the information
+/// we end up storing.
+public struct CommandEventGraphTarget: Codable, Hashable {
+    public init(
+        target: CommandEventTarget,
+        project: CommandEventProject
+    ) {
+        self.target = target
+        self.project = project
+    }
+
+    public init(
+        _ graphTarget: GraphTarget
+    ) {
+        target = CommandEventTarget(graphTarget.target)
+        project = CommandEventProject(graphTarget.project)
+    }
+
+    public let target: CommandEventTarget
+    public let project: CommandEventProject
```

Comment:
What about having the properties defined at the top of the `struct` to be consistent with the codebase? I'd do that with the new types introduced.